src/System.ServiceModel.Primitives/tests/ServiceModel/DuplexClientBaseTest.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+using Xunit;
+
+public class DuplexClientBaseTest
+{
+    [Fact]
+    public static void DuplexClientBase_Ctor_Initializes_State()
+    {
+        InstanceContext context = new InstanceContext(new WcfDuplexServiceCallback());
+        Binding binding = new NetTcpBinding();
+        EndpointAddress endpoint = new EndpointAddress(Endpoints.Tcp_NoSecurity_Callback_Address);
+        MyDuplexClientBase<IWcfDuplexService> duplexClientBase = new MyDuplexClientBase<IWcfDuplexService>(context, binding, endpoint);
+
+        Assert.Equal<EndpointAddress>(endpoint, duplexClientBase.Endpoint.Address);
+        Assert.Equal<CommunicationState>(CommunicationState.Created, duplexClientBase.State);
+
+        duplexClientBase.Abort();
+    }
```

Comment:
Question only: do you recommend we Abort or Close every channel, channelFactory or ClientBase we create?  And if so, should this be in a finally block so it executes even if the asserts fire?  I would be inclined to put them in using statements, but I believe WCF has the ability to throw from the Dispose.  Though it might be reasonable to let the Dispose throw and let xunit report it.  Thoughts @iamjasonp ?